torchaudio/prototype/models/conformer_wav2vec2.py

```diff
@@ -0,0 +1,298 @@
+from typing import List, Optional, Tuple, Union
+
+from torch import nn, Tensor
+from torch.nn import Module
+from torchaudio.models import Wav2Vec2Model
+from torchaudio.models.conformer import ConformerLayer
+from torchaudio.models.rnnt import _TimeReduction
+from torchaudio.models.wav2vec2 import components
+
+
+class TimeReduction(Module):
+    """Extract features from input. Consists of time reduction and linear layer.
+
+    Args:
+        stride (int): number of frames to merge for the output frame
+        input_dim (int): input dimension of the tensor
+        output_dim (int): output dimension of the tensor
```

Comment:
the order of the arguments can be same as the one in `_get_conformer_feature_extractor` for consistency